### PR TITLE
Feature add weekday temporal kernel

### PR DESCRIPTION
**Pull request by nl5887**
This feature will add the temporal kernel for weekday (day of week). When added, DOW support can be add to datafusion.

### CODE REVIEW

**Review by tustvold — COMMENTED** [review]
Could we get a test or two of this functionality please

File: arrow/src/compute/kernels/temporal.rs [hunk 1]
```diff
@@ -450,6 +479,19 @@ mod tests {
...
 
+    #[test]
+    fn test_temporal_array_date64_weekday() {
+        //1514764800000 -> 2018-01-01
+        //1550636625000 -> 2019-02-20
+        let a: PrimitiveArray<Date64Type> =
+            vec![Some(1514764800000), None, Some(1550636625000)].into();
+
+        let b = weekday(&a).unwrap();
+        assert_eq!(1, b.value(1));
```
Comment (alamb) [inline arrow/src/compute/kernels/temporal.rs:490]: Screen Shot 2022-06-16 at 4 08 52 PM

Screen Shot 2022-06-16 at 4 09 47 PM
👍

**Review by alamb — APPROVED** [review]
Looks good -- thank you for the contribution @nl5887 !

**Comment by alamb** [comment]
Looks like at least one test need to be updated:

https://github.com/apache/arrow-rs/runs/6924858466?check_suite_focus=true

Screen Shot 2022-06-16 at 4 11 00 PM

**Comment by tustvold** [comment]
Tests appear to be failing?

**Comment by nl5887** [comment]
> Looks like at least one test need to be updated:
> 
> https://github.com/apache/arrow-rs/runs/6924858466?check_suite_focus=true
> 
> Screen Shot 2022-06-16 at 4 11 00 PM

Thanks, updated the test. Any opinions if we should use the ISO 8601 (similar to chrono-rs) week days (starting at Monday) or the US (starting at Sunday)? Currently it is using ISO (https://docs.rs/chrono/latest/src/chrono/lib.rs.html#646).

**Comment by alamb** [comment]
> Thanks, updated the test. Any opinions if we should use the ISO 8601 (similar to chrono-rs) week days (starting at Monday) or the US (starting at Sunday)? Currently it is using ISO (https://docs.rs/chrono/latest/src/chrono/lib.rs.html#646).

I don't have any particular opinion -- it is likely that whatever we pick will be wrong for some usecases. The best long term is likely to allow the user to specify what behavior they want but I think we can do that in a follow on PR if anyone needs it

File: arrow/src/compute/kernels/temporal.rs [hunk 2]
```diff
@@ -211,6 +211,34 @@ where
     Ok(b.finish())
 }
 
+/// Extracts the day of week of a given temporal array as an array of integers
```
Comment (alamb) [inline arrow/src/compute/kernels/temporal.rs:214]:
```suggestion
/// Extracts the day of week of a given temporal array as an array of integers
/// Monday is 0, Tuesday is 1, etc.
```

Comment (alamb) [reply arrow/src/compute/kernels/temporal.rs:214]: proposed in https://github.com/apache/arrow-rs/pull/1894

**Pull request merged**